### PR TITLE
[release-4.8] OCPBUGS-3554: services: log errors and don't panic if the service doesn't exist

### PR DESCRIPTION
**Pull request by kyrtapz**
This is a cherry-pick of https://github.com/ovn-org/ovn-kubernetes/pull/2327 which never made it to 4.8.

/assign @dcbw

### CODE REVIEW

File: go-controller/pkg/ovn/controller/services/services_controller.go [hunk 1]
```diff
@@ -243,8 +243,12 @@ func (c *Controller) syncServices(key string) error {
 	if err != nil || !util.ServiceTypeHasClusterIP(service) || !util.IsClusterIPSet(service) {
 		err = deleteVIPsFromAllOVNBalancers(vipsTracked, name, namespace)
 		if err != nil {
-			c.eventRecorder.Eventf(service, v1.EventTypeWarning, "FailedToDeleteOVNLoadBalancer",
-				"Error trying to delete the OVN LoadBalancer for Service %s/%s: %v", name, namespace, err)
+			// If the service wasn't found, don't panic sending an
```
Comment (tssurya) [inline go-controller/pkg/ovn/controller/services/services_controller.go:246]: ooo great so that's why the panic huh...

**Review by tssurya — APPROVED** [review]

**Review by trozet — APPROVED** [review]
/label backport-risk-assessed

**Pull request merged**